Hometask2v2

Khabutdinov-Arslan opened the pull request:
New branch from most recent hometask1 version

habars2000 committed:
Class decomposition

habars2000 committed:
DIP fixes and other

habars2000 committed:
Lambda based hadling

habars2000 committed:
Lambda based event handling

habars2000 committed:
Added unit tests for handlers and helper functions

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/SmartHome.java:
```diff
@@ -3,13 +3,20 @@
     }
 
+    @Override
+    public void execute(Action action){
```
Дом тоже можно обрабатывать.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/SmartHomeHelpers.java:
```diff
@@ -0,0 +1,36 @@
+package ru.sbt.mipt.oop;
+
+public class SmartHomeHelpers {
+    static boolean isHallDoor(SmartHome smartHome, String id){
```
Это на удаление, теперь работа с домом только через execute.

a-ivanov reviewed (commented):
Удалите все методы с доступом к внутренним объектам дома.

a-ivanov reviewed (changes requested):
Доделайте.

habars2000 committed:
Replaced HomeComponent with actionable
Classes are split to packages
Action is executed in handler instead of returning it
Action is applied to the object itself before passing to children
Tests and helpers are refactored using actions
SmartHome is created in tests instead of reading it
HallDoorHandler now just turns off the lights

kieraCurtis commented on src/test/java/ru/sbt/mipt/oop/EventHandlersTest.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Test
+    public void HandleDoorOpenTest(){
```
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

a-ivanov commented on src/main/java/ru/sbt/mipt/oop/components/Room.java:
```diff
@@ -0,0 +1,43 @@
+        this.name = name;
+    }
+
+    public Collection<Light> getLights() {
```
Удалите.

habars2000 committed:
Delete getlights and getdoors

habars2000 committed:
Delete getlights and getdoors
